qiskit_experiments/hamiltonian_tomography/cr_hamiltonian_analysis.py

```diff
@@ -0,0 +1,354 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""
+Cross resonance Hamiltonian tomography analysis.
+"""
+
+from typing import List, Dict, Any, Union
+
+import numpy as np
+
+from qiskit_experiments.analysis import CurveAnalysis, SeriesDef, CurveAnalysisResult
+from qiskit_experiments.analysis.data_processing import expectation_value
+from qiskit_experiments.analysis.utils import get_opt_value, get_opt_error, frequency_guess
+from qiskit_experiments.exceptions import AnalysisError
+
+
+# pylint: disable = invalid-name
+def oscillation_x(x: np.ndarray, px: float, py: float, pz: float, b: float):
+    """Fit function for x basis oscillation."""
+    omega = np.sqrt(px ** 2 + py ** 2 + pz ** 2)
+    return (
+        -pz * px + pz * px * np.cos(omega * x) + omega * py * np.sin(omega * x)
+    ) / omega ** 2 + b
+
+
+# pylint: disable = invalid-name
+def oscillation_y(x: np.ndarray, px: float, py: float, pz: float, b: float):
+    """Fit function for y basis oscillation."""
+    omega = np.sqrt(px ** 2 + py ** 2 + pz ** 2)
+    return (pz * py - pz * py * np.cos(omega * x) - omega * px * np.sin(omega * x)) / omega ** 2 + b
+
+
+# pylint: disable = invalid-name
+def oscillation_z(x: np.ndarray, px: float, py: float, pz: float, b: float):
+    """Fit function for z basis oscillation."""
+    omega = np.sqrt(px ** 2 + py ** 2 + pz ** 2)
+    return (pz ** 2 + (px ** 2 + py ** 2) * np.cos(omega * x)) / omega ** 2 + b
+
+
+class ContinueOuterLoop(Exception):
+    """A fake exception to continue outer loop."""
+
+    pass
+
+
+class CRHamiltonianAnalysis(CurveAnalysis):
```

Comment:
Can you add a function `_evaluate_quality` that lets us know if the fit is good?